modules/core/src/smithy4s/http/internals/PathEncode.scala

```diff
@@ -19,14 +19,12 @@ package smithy4s.http.internals
 import smithy4s.internals.Hinted
 
 trait PathEncode[A] { self =>
-  def encode(sb: StringBuilder, a: A): Unit
-  def encodeGreedy(sb: StringBuilder, a: A): Unit
+  def encode(a: A): List[String]
```

Comment:
Not sure about the exact collection, Chain would've been nice here because these will be concatenated, but at the same time paths aren't really often long enough for this to make a difference (or are they?).

Reply:
Could keep it mutable a use a ListBuilder

Reply:
oh I see what you mean, I'll just replace SB with that.

Reply:
I'm not convinced there's a major benefit: for small collections it's really negligible, whereas the code is quite a bit more complex. Implemented this using ArrayBuilder (there's no ListBuilder it seems): [`34d5344` (#74)](https://github.com/disneystreaming/smithy4s/pull/74/commits/34d53448adc3c2b45a39783fba310613138eff10) but the commit can easily be reverted if you agree

Reply:
there's ListBuffer, but the takeaways are the same 😅

Reply:
@kubukoz yeah okay, I can agree that it's likely for paths to be small enough that mutability is overkill. Go ahead and revert